lightning/src/ln/channelmanager.rs

```diff
@@ -5042,13 +5156,12 @@ impl<Signer: Sign, M: Deref, T: Deref, K: Deref, F: Deref, L: Deref> Writeable f
 			pending_payment.write(writer)?;
 		}
 
-		let pending_outbound_payments = self.pending_outbound_payments.lock().unwrap();
-		(pending_outbound_payments.len() as u64).write(writer)?;
-		for session_priv in pending_outbound_payments.iter() {
-			session_priv.write(writer)?;
-		}
+		(0 as u64).write(writer)?; // for backwards compat with the old format of pending_outbound_payments
```

Comment:
Hmm, so this is a "backwards incompatibility". I dunno if we're *super* worried about it, but practically this means "old clients will not create *any* events for all payments initiated by new clients". In general, if its easy, I think we'd prefer to avoid things like this. One easy way to do that here would be to write out the full set of session privs but not the mpp ids until the TLV entry.

Reply:
> One easy way to do that here would be to write out the full set of session privs but not the mpp ids until the TLV entry.

Went with this